File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> obj;
        obj['I'] = 1;obj['V']=5;obj['X']=10;obj['L']=50;obj['C']=100;obj['D']=500;obj['M']=1000;
        s += "I";
        int i = 0;
        int res = 0;
        while (s[i+1] != '\0') {
            if (obj[s[i]] >= obj[s[i+1]])res += obj[s[i]];
            else {res += obj[s[i+1]] - obj[s[i]];i++;}
            i++;
        }
        return res;
    }
};

File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        dic = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        string = s + 'I'
        curr = 0
        res = 0
        length = len(string)
        while (curr < length-1) :
            if dic[string[curr]] >= dic[string[curr+1]]:res += dic[string[curr]]
            else :
                res += dic[string[curr+1]] - dic[string[curr]]
                curr+=1
            curr+=1
        return res